libp2p/errors.nim

```diff
@@ -9,6 +9,8 @@ type
   # Base exception type for libp2p
   LPError* = object of CatchableError
 
+  LPDefect* = object of CatchableError
```

Comment:
I find this a bit confusing
I get the idea of "a libp2p non recoverable error may not be an app non recoverable error, so let's make a fake defect"

But that's more confusing than anything, imo

Reply:
I totally agree, will change it use `LPError` instead. Not sure what the line of reasoning was here. With the callbacks, the calling applications can re-raise a defect if needed.